machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,21 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// DiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	DiskSettings DiskSettings `json:"diskSettings,omitempty"`
+	// CachingType specifies the caching requirements.
+	// Possible values include: 'None', 'ReadOnly', 'ReadWrite'. When empty defaults to 'None'.
```

Comment:
Recall the bits about the default being subject to change in a future release.

Reply:
needs validation tags or to be built as an enum

Reply:
@Fedosin  can you update this to match below
```suggestion
	// Possible values include: 'None', 'ReadOnly', 'ReadWrite'.
	// Empty value means no opinion and the platform chooses a default, which is subject to change over
	// time. Currently the default is `None`.
```